### PR TITLE
NIAD-402: Using correct cipher in  NHS segment and added inbound UAT …

### PR DESCRIPTION
**Pull request by grzegorz-robak**
…RECEP tests

### CODE REVIEW

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/inbound/RecepProducerService.java [hunk 1]
```diff
@@ -119,8 +120,19 @@ private RecepHeader mapToInterchangeHeader(Interchange interchange) {
...
+        var cypher = message.getHealthAuthorityNameAndAddress().getIdentifier();
+        var gpCode = findGpCode(message);
+        return new RecepNationalHealthBody(cypher, gpCode);
+    }
+
+    private String findGpCode(Message message) {
+        try {
+            return message.getTransactions().get(0).getGpNameAndAddress().getIdentifier();
+        } catch (MissingSegmentException e) {
+            return "9999"; //default if NAD+GP segment is missing in first transaction
```
Comment (jaroslaw-wojciechowski) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/inbound/RecepProducerService.java]: should be static final

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/Message.java [hunk 2]
```diff
@@ -54,4 +55,16 @@ public String toString() {
...
+    public String findFirstGpCode() {
+        return getTransactions()
+            .stream()
+            .limit(1)
+            .map(transaction -> transaction.extractOptionalSegment(GpNameAndAddress.KEY_QUALIFIER))
+            .flatMap(Optional::stream)
+            .map(GpNameAndAddress::fromString)
+            .map(GpNameAndAddress::getIdentifier)
+            .findFirst()
+            .orElse("9999");
```
Comment (jaroslaw-wojciechowski) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/Message.java]: should be static final

**Review by jaroslaw-wojciechowski — CHANGES_REQUESTED** [review]

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/inbound/RecepProducerService.java (hunk 1 above)
Comment (grzegorz-robak) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/inbound/RecepProducerService.java]: Just did that :)

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/Message.java (hunk 2 above)
Comment (grzegorz-robak) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/Message.java]: Just did that :)

**Review by jaroslaw-wojciechowski — APPROVED** [review]

**Review by mattd-kainos — APPROVED** [review]

**Pull request merged**